ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -134,6 +134,16 @@ class CKTypedComponentAction : public CKTypedComponentActionBase {
   }
 
   /**
+   Allows demoting an action to a simpler action while supplying defaults for the values that won't be passed in.
+   */
+  template<typename... U>
+  static CKTypedComponentAction<T...> demotedFrom(CKTypedComponentAction<T..., U...> action, U... defaults) {
+    return CKTypedComponentAction<T...>::actionFromBlock(^(CKComponent *sender, T... args) {
```

Comment:
Alright, my one concern here is if the type is a reference, the capture will be garbage at execution time, which sucks. I think this is OK for now, but mind following up and seeing if you can add a static assert or perhaps augment this with a decayed type to force references to values?